### PR TITLE
Amend parami protocol

### PR DESCRIPTION
**Pull request by hellotrongo**
#160

### CODE REVIEW

**Comment by alxs** [comment]
Thanks for the amendment @hellotrongo. As I mentioned, could you please include deliverables 0a-d as in the [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules) in all milestones, and 0e at least in the last one? Also I think the price should be split evenly across all milestones. You're quoting a similar workload for each of them and if anything, it seems to me the last milestone would be the most extensive one.

File: applications/parami-protocol.md [hunk 1]
```diff
@@ -150,25 +150,57 @@ Yanping YANG
...
+| 2a. | Substrate chain | Completed the bridge development between ETH ERC20 and Parami chain. Please refer to https://github.com/parami-protocol/parami/tree/main/pallets/bridge  |
+| 2b. | Substrate chain | Complete NFT pallets, please refer to https://github.com/parami-protocol/parami/tree/main/pallets/nft |
+| 3. | Parami wallet	 | Parami wallet has supported Telegram and WeChat. Users can login to Parami wallet in the social software to complete the binding of social ID and DID identity, implement token transfer in/out and transaction query functions. Please refer to: https://github.com/parami-protocol/wallet_server |
+| 4. | ad3-maker | encode/decode the avatar of Telegram or Wechat users, and create the binding of DID and social user's identity image. Social users can transfer token through the avatar. Please refer to: https://github.com/parami-protocol/ad3-marker |
+
+### Milestone  śīla(July 8 - August 8 Partially Completed)
```
Comment (alxs) [inline applications/parami-protocol.md:173]: Please also number your milestones.

```suggestion
### Milestone 1 - śīla(July 8 - August 8 Partially Completed)
```

**Comment by hellotrongo** [comment]
Hello @alxs , thanks for the suggestions. I've updated the pr.

**Comment by DorianRust** [comment]
Hi @alx, sorry to disturb you again. The cost has been adjusted. Besides this part, I believe I have amended everything based on your suggestion. What else did I miss? Thank you.

**Comment by alxs** [comment]
Hi Dorian. Could you please include _the same_ deliverables 0a-d as in the template in every milestone? Eventually you can change the wording etc. But they should be in the spirit of those in the template i.e. 0a is the license, 0b is documentation etc. These are "meta-deliverables" which is why they are labelled with 0. Your other deliverables should each have their own number and can be split into a, b, c, if they clearly belong together or are part of the same component.

**Comment by DorianRust** [comment]
Hi @alxs, thanks for your suggestions. I have revised the PR accordingly.

**Comment by alxs** [comment]
Thanks, could you label the deliverables according to what they are? I.e. instead of 10 times "substrate chain" please name the pallet(s)/interface that you will develop. The rest looks good to me.

**Comment by hellotrongo** [comment]
Hi @alxs, Hi, please kindly find the amended version. Thanks for all the suggestions and support.

File: applications/parami-protocol.md [hunk 2]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+| 0e. | Article | We will publish an achieved that explains what was done). |
+| 1. | Pallet(s)/onchain | Develop on-chain governance logic to build a more decentralized platform. |
+| 2. | AD3-Maker | encode/decode the avatar of Telegram or Wechat users, and create the binding of DID and social user's identity image. Social users can transfer token through the avatar. Please refer to: https://github.com/parami-protocol/ad3-marker |
+| 3. | Parami AMM | Implement Parami AMM, support AD3, Social coin. |
+| 4. | Parami Wallet | Parami wallet supports blind signature algorithm, which are used to obtain reward evidence for users; the data layer supports homomorphic encryption algorithms to provide privacy protection of users' data.  |
+| 5. | Parami Wallet | Integrate zero-knowledge proof technology for users' private data update, verifiable computaion of reward amount and the implementation of anonymous transactions to fully archive privacy protection. |
+| 6. | Parami Wallet | Wallet development, optimization of new functions, improve the smoothness of users using wallets, and enhance users' experience; (finished)|
```
Comment (alxs) [inline applications/parami-protocol.md:205]: Could you do the same here please?

File: applications/parami-protocol.md [hunk 3]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article or medium that explains what was done as part of the grant). |
+| 1. | Pallet(s)/did | Build Parami ParaChain based on Substrate3.0.0, the core modules that have been developed include DID(Decentralized Identity) pallet, Decentralised identity is an emerging concept that gives back control of identity to consumers. |
+| 2. | Pallet(s)/bridge | Completed the bridge development between ETH ERC20 and Parami chain. Please refer to https://github.com/parami-protocol/parami/tree/main/pallets/bridge  |
+| 3. | Pallet(s)/nft | Complete NFT pallets, please refer to https://github.com/parami-protocol/parami/tree/main/pallets/nft |
+| 4. | Testnet | Setup and launch Parami testnet, which has been running stably for about a month. Parami testnet has has become a parachain to Rococo. Please refer to (https://polkadot.js.org/apps/?rpc=wss://rococo-rpc.polkadot. io#/parachains/parathreads). |
```
Comment (alxs) [inline applications/parami-protocol.md:168]: We only fund deliverables that can be reused others in the ecosystem. Could you remove this?

File: applications/parami-protocol.md [hunk 4]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation to help developers to use the wallet |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality for wallet. |
+| 0e. | Article | We will publish an medium that explains what was done as part of the grant). |
+| 1. | Pallet(s)/offchain | Develop Offchain-worker to verify ads and incentives, rewards and withdrawals, etc.
+| 2. | Pallet(s)/socialcoin | Support Social coin and introduce NFT support for advertisers.
```
Comment (alxs) [inline applications/parami-protocol.md:183]: This "social coin" isn't described anywhere in the application.

**Comment by alxs** [comment]
Sorry @hellotrongo, I'm gonna go through it one more time. The new deliverables diverge considerably from the original application.

File: applications/parami-protocol.md [hunk 5]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation to help developers to integrate Parami DID into their project |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an article or medium that explains what was done as part of the grant). |
+| 1. | Pallet(s)/did | Build Parami ParaChain based on Substrate3.0.0, the core modules that have been developed include DID(Decentralized Identity) pallet, Decentralised identity is an emerging concept that gives back control of identity to consumers. |
```
Comment (alxs) [inline applications/parami-protocol.md:165]: Description doesn't match deliverable name.

File: applications/parami-protocol.md [hunk 6]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+| 2. | Pallet(s)/bridge | Completed the bridge development between ETH ERC20 and Parami chain. Please refer to https://github.com/parami-protocol/parami/tree/main/pallets/bridge  |
+| 3. | Pallet(s)/nft | Complete NFT pallets, please refer to https://github.com/parami-protocol/parami/tree/main/pallets/nft |
```
Comment (alxs) [inline applications/parami-protocol.md:167]: There's no mention of either of these anywhere else in the application, and to be honest I don't think we're interested to fund either of them (there are already plenty of implementations for both).

File: applications/parami-protocol.md [hunk 7]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+* **Costs:** 0.33 BTC
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation to help developers to use the wallet |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 70%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality for wallet. |
+| 0e. | Article | We will publish an medium that explains what was done as part of the grant). |
+| 1. | Pallet(s)/offchain | Develop Offchain-worker to verify ads and incentives, rewards and withdrawals, etc.
```
Comment (alxs) [inline applications/parami-protocol.md:182]: Please describe precisely what the implemented functionality will be.

File: applications/parami-protocol.md [hunk 8]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 80%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality. |
+| 0e. | Article | We will publish an achieved that explains what was done). |
+| 1. | Pallet(s)/onchain | Develop on-chain governance logic to build a more decentralized platform. |
+| 2. | AD3-Maker | encode/decode the avatar of Telegram or Wechat users, and create the binding of DID and social user's identity image. Social users can transfer token through the avatar. Please refer to: https://github.com/parami-protocol/ad3-marker |
+| 3. | Parami AMM | Implement Parami AMM, support AD3, Social coin. |
```
Comment (alxs) [inline applications/parami-protocol.md:202]: No description of the AMM in the rest of the application. Also here, plenty of implementations exist and you can likely reuse any of them.

Comment (alxs) [inline applications/parami-protocol.md:202]: No description of the AMM in the rest of the application. Also here, plenty of implementations exist and you can likely reuse any of them.

File: applications/parami-protocol.md [hunk 9]
```diff
@@ -146,29 +146,65 @@ Yanping YANG
...
+* **Costs:** 0.34 BTC 
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide inline documentation of the code and API documentation |
+| 0c. | Testing Guide | The code will have unit-test coverage (min. 80%) to ensure functionality and robustness. In the guide we will describe how to run these tests |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality. |
+| 0e. | Article | We will publish an achieved that explains what was done). |
+| 1. | Pallet(s)/onchain | Develop on-chain governance logic to build a more decentralized platform. |
```
Comment (alxs) [inline applications/parami-protocol.md:200]: Governance modules already exist. We won't fund "governance logic to build a more decentralized platform".

**Review by alxs — CHANGES_REQUESTED** [review]

File: applications/parami-protocol.md (hunk 3 above)
Comment (hellotrongo) [reply applications/parami-protocol.md:168]: remove Testnet?

**Comment by hellotrongo** [comment]
Hi @alxs , Thanks for your support along the way. Given that I received very informative suggestions on your side last time, I dived into the deliverables and reconstructed the whole stuff. The current version is more concise because I removed some unnecessary parts based on your suggestions. If any further advice, please feel free to contact me. Thanks.

**Comment by alxs** [comment]
As you will. The delivery would still be overdue with the updated duration, could you calculate it from the date your original application was accepted or just include an ETA? Thanks.

**Comment by hellotrongo** [comment]
Hi @alxs, thanks for the reminder. The deadline is updated to until December, which would be 12 months since the original application was accepted.

**Comment by alxs** [comment]
@hellotrongo thanks. That's a very long timeframe, I personally think in that case and seeing as your roadmap doesn't seem to be exactly aligned with the deliverables, it would make more sense for everyone involved to terminate the current grant and that you submit a new application. Let's see what the rest of the committee thinks.

**Comment by hellotrongo** [comment]
Hi @alxs. It’s basically same, I just add a little bit more content. And we already finished some content of grant, why do you terminate my grant?

**Comment by hellotrongo** [comment]
Hi @alxs, We didn’t change the basic structure, only to make some minor adjustments and a few additions. Given the only major change is time duration alone, do we have to submit a new application? We’d be happy to re-apply, while it would be better if we can make amendments to the original one.

File: applications/parami-protocol.md [hunk 10]
```diff
@@ -146,29 +146,31 @@ Yanping YANG
 ## Development Roadmap :nut_and_bolt:
 
 ### Overview
-* **Total Estimated duration:** 4 months
+* **Total Estimated duration:** 6 months
```
Comment (Noc2) [inline applications/parami-protocol.md:149]: @hellotrongo  Is the total **Total Estimated duration:** 6 months?

File: applications/parami-protocol.md [hunk 11]
```diff
@@ -146,29 +146,31 @@ Yanping YANG
...
 * **Full-time equivalent (FTE):** 3 FTE
 * **Total Costs:** 1 BTC
 
 ### Milestone 1 — The Parami Protocol
-* **Estimated duration:** 4 month
+* **Estimated duration:** 12 month
```
Comment (Noc2) [inline applications/parami-protocol.md:154]: Or 12 months?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Do the 6 or 12 month start with the amendment or do you think the 6 or 12 months start with the original contract?

File: applications/parami-protocol.md (hunk 11 above)
Comment (hellotrongo) [reply applications/parami-protocol.md]: Hi @Noc2, sorry it’s 12 months. I have aligned the duration. We spent quite a lot time exploring Zero Knowledge Proof and its adoption to advertising at the beginning of development, including studying cryptography with researchers in academia, which extended the duration.

**Comment by Noc2** [comment]
Thanks. In this case, my recommendation is also to cancel the current contract and potentially sign a new contract at the end of the year. A lot of things might change in half a year, especially in the crypto space. For example other teams might already have implemented the features, or you might no longer want to work on this, etc.

File: applications/parami-protocol.md (hunk 9 above)
Comment (hellotrongo) [reply applications/parami-protocol.md]: hi @alxs ,  We spent quite a lot time exploring Zero Knowledge Proof and its adoption to advertising at the beginning of development, including studying cryptography with researchers in academia, which extended the duration. And we also finished some milestone in the last application.

**Comment by hellotrongo** [comment]
Hi @alxs, After some careful evaluation with the tech team, we found that 8-month duration is doable. I have amended the duration here. Thanks.

**Comment by Tailalat** [comment]
> #160

**Comment by alxs** [comment]
@hellotrongo it seems the rest of the committee isn't convinced by the amendment either, despite the shortened duration. I'm closing the PR and putting the grant forward for termination again (which we shouldn't have undone without negotiating the amendment first, apologies for the oversight). As mentioned, feel free to apply again with your current roadmap.

**Pull request closed without merging**